File: typical90/018/test/main.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,a,b) for(int i=a;i<b;i++)
#define rrep(i,a,b) for(int i=a;i<=b;i++)
#define drep(i,a,b) for(int i=a;i>=b;i--)
#define fore(i,a) for(auto &i:a)
typedef long long ll;
long double PI = 3.14159265358979;
const ll INF = 1000000000000000000LL;
long double t,l,x,y,dx,dy,dz;

long double ans(long double e) {
  long double d1, radian;
  dx = 0;
  dy = -(l/2.0)*sin(e/t*2.0*PI);
  dz = (l/2.0) - (l/2.0)*cos(e/t*2.0*PI);
  d1 = sqrt((dx-x)*(dx-x)+(dy-y)*(dy-y));
  radian = atan2(dz,d1);
  return radian * 180.0L / PI;
};

int main() {
  int q; cin >> t;
  cin >> l >> x >> y;
  cin >> q;

  rep(i,0,q) {
    double e; cin >> e;
    printf("%.12Lf\n",ans(e));
  }
  return 0;
}

File: typical90/018/test/main.py
import math

def distance_point(x1, y1, x2, y2):
    d = math.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)
    return d

T = int(input())
L, X, Y = map(int,input().split())
Q = int(input())

for q in range(Q):
    e = int(input())
    radi = math.radians((360/T)*e-90)
    y = - L/2 * math.cos(radi)
    z = L/2 + L/2 * math.sin(radi)

    d = distance_point(0, y, X, Y)
    print(math.degrees(math.atan2(z, d)))